code/datums/components/pellet_cloud.dm

```diff
@@ -17,20 +17,34 @@
 */
 
 /datum/component/pellet_cloud
-	var/projectile_type /// What's the projectile path of the shrapnel we're shooting?
-
-	var/num_pellets /// How many shrapnel projectiles are we responsible for tracking? May be reduced for grenades if someone dives on top of it. Defined by ammo casing for casings, derived from magnitude otherwise
-	var/radius = 4 /// For grenades/landmines, how big is the radius of turfs we're targeting? Note this does not effect the projectiles range, only how many we generate
-
-	var/list/pellets = list() /// The list of pellets we're responsible for tracking, once these are all accounted for, we finalize.
-	var/list/targets_hit = list() /// An associated list with the atom hit as the key and how many pellets they've eaten for the value, for printing aggregate messages
-	var/list/bodies /// For grenades, any /mob/living's the grenade is moved onto, see [/datum/component/pellet_cloud/proc/handle_martyrs()]
-	var/list/purple_hearts /// For grenades, tracking people who die covering a grenade for achievement purposes, see [/datum/component/pellet_cloud/proc/handle_martyrs()]
-
-	var/terminated /// how many pellets ranged out without hitting anything
-	var/hits /// how many pellets impacted something
-
-	var/mob/living/shooter /// for if we're an ammo casing being fired
+	/// What's the projectile path of the shrapnel we're shooting?
+	var/projectile_type
+
+	/// How many shrapnel projectiles are we responsible for tracking? May be reduced for grenades if someone dives on top of it. Defined by ammo casing for casings, derived from magnitude otherwise
+	var/num_pellets
+	/// For grenades/landmines, how big is the radius of turfs we're targeting? Note this does not effect the projectiles range, only how many we generate
+	var/radius = 4
+
+	/// The list of pellets we're responsible for tracking, once these are all accounted for, we finalize.
+	var/list/pellets = list()
+	/// An associated list with the atom hit as the key and how many pellets they've eaten for the value, for printing aggregate messages
+	var/list/targets_hit = list()
+	/// For grenades, any /mob/living's the grenade is moved onto, see [/datum/component/pellet_cloud/proc/handle_martyrs()]
+	var/list/bodies
+	/// For grenades, tracking people who die covering a grenade for achievement purposes, see [/datum/component/pellet_cloud/proc/handle_martyrs()]
+	var/list/purple_hearts
```

Comment:
Couldn't the `targets_hit` get deleted while a reference to them remains here?
You could just save their name and locations, since that's what the visible name needs.